src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -38,10 +40,35 @@
  */
 public class InternalSettingsPreparer {
 
+    public static final String SECRET_PROMPT_VALUE = "__prompt:secret__";
```

Comment:
sorry :).. good with what? the original (i.e. `__prompt:secret__`) or the dollar sign option (i.e. `${prompt::text}`)?